### DISCUSSION TITLE
Azure SQL server connection using Azure Directory password

### DISCUSSION THREAD

**Discussion by Elucidator2197** [prisma/prisma#24697] [Q&A] [answered]
### Question

Is it possible to connect to a Azure SQL server using a Active Directory password ? If so can you give an example because i am facing P1000 Error

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @Elucidator2197 👋 

 It is possible to connect to an Azure SQL server using an Active Directory password. You can use a connection string that includes the `integratedSecurity` parameter along with the `username` and `password` parameters.

Here is an example connection string for connecting to an Azure SQL server using an Active Directory password:
```
sqlserver://localhost:1433;database=sample;integratedSecurity=true;username=prisma;password=aBcD1234;trustServerCertificate=true;
```
See this [section](https://www.prisma.io/docs/orm/overview/databases/sql-server#using-integrated-security-windows-only) of the documentation.

Make sure to replace `localhost:1433, sample, prisma, and aBcD1234` with your actual server address, database name, Active Directory username, and password respectively. If you are facing the P1000 error, it indicates that the authentication failed. Ensure that the provided database credentials are correct and valid for the database server you are trying to connect to. For Microsoft SQL Server, you must [escape special characters](https://www.prisma.io/docs/orm/overview/databases/sql-server#connection-details) in any part of your connection string.

**Reply by Elucidator2197** [reply]
i did try it 
so first i tried 
DATABASE_URL='sqlserver://serverName.database.windows.net:1433;database=databaseNameintegratedSecurity=true;username={email@homeoffice.com-pany.com};password={Password@1};trustServerCertificate=true;'
it gave me an error
Error: Cannot open server "homeoffice.com-pany.com" requested by the login.  The login failed.


then i tried 
DATABASE_URL='sqlserver://serverName.database.windows.net:1433;database=databaseName;integratedSecurity=true;username={email@homeoffice.com-pany.com@serverName);password={Password@1};trustServerCertificate=true;'

it gave me an error Error: P1000
Authentication failed against database server

Any suggestions ?
because i know credentials are correct

**Reply by ValentinChion** [reply]
Anyone has answers ? I have the exact same problem
